src/overlays/actors/ovl_En_Anubice/z_en_anubice.c

```diff
@@ -53,35 +63,439 @@ static ColliderCylinderInit D_809B22D0 = {
+}
+
+void EnAnubice_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnAnubice* this = THIS;
+    EnAnubiceTag* temp_v1;
+
+    Collider_DestroyCylinder(globalCtx, &this->col);
+
+    if (this->actor.params != 0) {
+        if (this->actor.parent) {}
+
+        temp_v1 = (EnAnubiceTag*)this->actor.parent;
+        if (temp_v1 != NULL && temp_v1->actor.update != NULL) {
+            temp_v1->anubis = NULL;
+        }
+    }
+}
+
+void EnAnubice_FindFlameCircles(EnAnubice* this, GlobalContext* globalCtx) {
+    Actor* currentProp;
+    s32 flameCirclesFound;
+
+    if (this->isMirroringLink) {
+        if (!this->hasSearchedForFlameCircles) {
+            flameCirclesFound = 0;
+            currentProp = globalCtx->actorCtx.actorLists[ACTORCAT_PROP].head;
+            while (currentProp != NULL) {
+                if (currentProp->id != ACTOR_BG_HIDAN_CURTAIN) {
+                    currentProp = currentProp->next;
+                } else {
+                    this->flameCircles[flameCirclesFound] = (BgHidanCurtain*)currentProp;
+                    // ☆☆☆☆☆ How many fires? ☆☆☆☆☆
+                    osSyncPrintf(VT_FGCOL(GREEN) "☆☆☆☆☆ 火は幾つ？ ☆☆☆☆☆ %d\n" VT_RST, flameCirclesFound);
+                    osSyncPrintf(VT_FGCOL(YELLOW) "☆☆☆☆☆ 火は幾つ？ ☆☆☆☆☆ %x\n" VT_RST,
+                                 this->flameCircles[flameCirclesFound]);
+                    if (flameCirclesFound < 4) {
+                        flameCirclesFound++;
+                    }
+                    currentProp = currentProp->next;
+                }
+            }
+            this->hasSearchedForFlameCircles = true;
+        }
+        this->actor.flags |= 1;
+        this->actionFunc = EnAnubice_SetupIdle;
+    }
+}
+
+void EnAnubice_SetupIdle(EnAnubice* this, GlobalContext* globalCtx) {
+    f32 lastFrame = Animation_GetLastFrame(&D_06000F74);
+
+    Animation_Change(&this->skelAnime, &D_06000F74, 1.0f, 0.0f, (s16)lastFrame, 0, -10.0f);
+
+    this->actionFunc = EnAnubice_Idle;
+    this->actor.velocity.x = this->actor.velocity.z = this->actor.gravity = 0.0f;
+}
+
+void EnAnubice_Idle(EnAnubice* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
+
+    SkelAnime_Update(&this->skelAnime);
+    Math_ApproachZeroF(&this->actor.shape.yOffset, 0.5f, 300.0f);
+    Math_ApproachF(&this->focusHeightOffset, 70.0f, 0.5f, 5.0f);
+
+    if (!this->isKnockedback) {
+        Math_SmoothStepToS(&this->actor.shape.rot.y, this->actor.yawTowardsPlayer, 5, 3000, 0);
+    }
+
+    if (this->actor.shape.yOffset > -2.0f) {
+        this->actor.shape.yOffset = 0.0f;
+
+        if (player->swordState != 0) {
+            this->actionFunc = EnAnubis_SetupShootFireball;
+        } else if (this->isLinkOutOfRange) {
+            this->actor.velocity.y = 0.0f;
+            this->actor.gravity = -1.0f;
+            this->actionFunc = EnAnubice_GoToInitPos;
+        }
+    }
+}
+
+void EnAnubice_GoToInitPos(EnAnubice* this, GlobalContext* globalCtx) {
+    f32 xzdist;
```

Comment:
with the recent-ish actor struct changes we refer to this as the actors "home" position. might be good to use that terminology here, but initial works just as well. i leave that up to you